test/maverick/consensus/replay.go

```diff
@@ -469,7 +469,16 @@ func (h *Handshaker) replayBlocks(
 			assertAppHashEqualsOneFromBlock(appHash, block)
 		}
 
-		appHash, err = sm.ExecCommitBlock(proxyApp.Consensus(), block, h.logger, h.stateStore, h.genDoc.InitialHeight)
+		appHash, err = sm.ExecCommitBlock(
```

Comment:
While the overall goal here is fine, adding random extra parameters to an existing exported API doesn't seem like a good choice—especially since the new arguments are all to serve a special case. I especially don't like the naked Boolean; that pattern has been a real hassle for ongoing maintenance in my experience.

I think a better approach here would be to add a new method, perhaps taking an argument struct. We can re-implement the existing method in terms of that, to avoid duplication, but at least then the usage will be more self-documenting.

Reply:
sure, will redo it.